compiler-plugin/src/main/kotlin/com/google/devtools/ksp/processing/impl/ResolverImpl.kt

```diff
@@ -356,20 +360,24 @@ class ResolverImpl(
     override fun mapToJvmSignature(declaration: KSDeclaration): String? = mapToJvmSignatureInternal(declaration)
 
     internal fun mapToJvmSignatureInternal(declaration: KSDeclaration): String? = when (declaration) {
-        is KSClassDeclaration -> resolveClassDeclaration(declaration)?.let { typeMapper.mapType(it).descriptor }
+        is KSClassDeclaration -> resolveClassDeclaration(declaration)?.let { typeMapper.mapType(it.defaultType).descriptor }
         is KSFunctionDeclaration -> resolveFunctionDeclaration(declaration)?.let {
             when (it) {
                 is FunctionDescriptor -> typeMapper.mapAsmMethod(it).descriptor
-                is PropertyDescriptor -> typeMapper.mapFieldSignature(it.type, it) ?: typeMapper.mapType(it).descriptor
+                is PropertyDescriptor -> typeMapper.mapPropertySignature(it)
                 else -> throw IllegalStateException("Unexpected descriptor type for declaration: $declaration")
             }
         }
-        is KSPropertyDeclaration -> resolvePropertyDeclaration(declaration)?.let {
-            typeMapper.mapFieldSignature(it.type, it) ?: typeMapper.mapType(it).descriptor
-        }
+        is KSPropertyDeclaration -> resolvePropertyDeclaration(declaration)?.let { typeMapper.mapPropertySignature(it) }
```

Comment:
Nit: it is exactly 120 line length, but keeping the same structure (and just rewrite `mapFieldSignature` with `mapPropertySignature`) would make the overall change look minimal.